### PR TITLE
Fixes #34555 - redesign scheduling step job wizard

### PR DESCRIPTION
**Pull request by MariaAga**
Screenshot from 2022-03-29 09-37-26
Screenshot from 2022-03-29 09-37-33
Screenshot from 2022-03-29 09-38-57

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,11 +21,11 @@
   },
   "devDependencies": {
     "@babel/core": "^7.7.0",
-    "@theforeman/builder": "^8.16.0",
-    "@theforeman/eslint-plugin-foreman": "^8.16.0",
-    "@theforeman/stories": "^8.16.0",
-    "@theforeman/test": "^8.16.0",
-    "@theforeman/vendor-dev": "^8.16.0",
+    "@theforeman/builder": "^10.1.0",
```
Comment (MariaAga) [inline package.json:24]: The core pr will be probably merged before this one https://github.com/theforeman/foreman/pull/9110

**Comment by MariaAga** [comment]
Missing: clearing function for the `Future execution` step datepickers ( probably buttons)

**Comment by adamruzicka** [comment]
Looks nice, the only thing I was surprised with was that you can select future or recurring execution and then submit the job without filling in any details. Not saying its wrong, I just didn't see it coming

File: package.json (hunk 1 above)
Comment (adamruzicka) [reply package.json:24]: The core PR was indeed merged before this one, resolving.

**Comment by MariaAga** [comment]
Not sure what a solution to the surprise you mentioned, but uses can see the pre-selected values at the next step and also in the review step
Re did this pr a bit, main changes are:
* Added a few visual dividers 
* moved the query type to the `type of execution` step
* Added clear buttons for the future execution step
* Made starts at/before mandatory
* Pre selected starts at to be now

**Comment by MariaAga** [comment]
@Ron-Lavi can you take a look?

**Comment by adamruzicka** [comment]
@MariaAga needs a rebase

@Ron-Lavi I'm ok with merging the smaller PRs on my own, but I'd really appreciate your input on this one

**Comment by jeremylenz** [comment]
@adamruzicka @MariaAga taking a look, stand by for review :)

File: webpack/JobWizard/steps/Schedule/ScheduleType.js [hunk 2]
```diff
@@ -1,28 +1,96 @@
...
+              scheduleType: SCHEDULE_TYPES.RECCURING,
+              repeatType: repeatTypes.daily,
+              repeatData: { at: new Date() },
+            }));
+            setValid(true);
+          }}
+          name="schedule-type-reccuring"
+          id="schedule-type-reccuring"
+          label={__('Recurring execution')}
+          body={__('Set up a reccuring execution for a job.')}
```
Comment (jeremylenz) [inline webpack/JobWizard/steps/Schedule/ScheduleType.js:77]:
```suggestion
          body={__('Execute the job on a repeating schedule.')}
```

File: webpack/JobWizard/JobWizard.js [hunk 3]
```diff
@@ -187,18 +191,80 @@ export const JobWizard = () => {
...
+                canJumpTo: isTemplate && valid.hostsAndInputs && valid.advanced,
+                enableNext:
+                  isTemplate &&
+                  valid.hostsAndInputs &&
+                  valid.advanced &&
+                  valid.schedule,
+              },
+            ]
+          : []),
+        ...(scheduleValue.scheduleType === SCHEDULE_TYPES.RECCURING
```
Comment (jeremylenz) [inline webpack/JobWizard/JobWizard.js:242]:
```suggestion
        ...(scheduleValue.scheduleType === SCHEDULE_TYPES.RECURRING
```

Recurring has two R's, not two C's.. should be changed everywhere ;)

File: webpack/JobWizard/steps/Schedule/ScheduleReccuring.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
+import React, { useEffect, useState } from 'react';
```
Comment (jeremylenz) [inline webpack/JobWizard/steps/Schedule/ScheduleReccuring.js:1]: file name spelling should be corrected `ScheduleRecurring.js`

File: webpack/JobWizard/steps/Schedule/ScheduleFuture.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      setValid(false);
+      setError(__("'Starts before' date must be after 'Starts at' date"));
+    } else if (new Date().getTime() >= new Date(startsBefore).getTime()) {
+      setValid(false);
+      setError(__("'Starts before' date must in the future"));
+    } else {
+      setValid(true);
+      setError(null);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (jeremylenz) [inline webpack/JobWizard/steps/Schedule/ScheduleFuture.js:44]: See previous comment about disabling this rule.

**Review by jeremylenz — COMMENTED** [review]
Thanks @MariaAga! Seems to work well; some comments below.

Not sure if this is related, but when I select Recurring and try to submit, I'm getting a console error:

```
Uncaught TypeError: repeatData.at.split is not a function
    at getCronLine (submit.js:39:23)
    at submit (submit.js:76:26)
    at onSave (JobWizard.js:303:9)
    at Wizard.onNext (Wizard.js:34:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237:1)
    at invokeGuardedCallback (react-dom.development.js:292:1)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:306:1)
    at executeDispatch (react-dom.development.js:389:1)
    at executeDispatchesInOrder (react-dom.development.js:414:1)
```

**Comment by MariaAga** [comment]
Thanks @jeremylenz , added a commit to address your comments

**Review by jeremylenz — APPROVED** [review]
Changes look good, thanks! (also Ruby test failures seem unrelated, they're about the template name change thing)

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @jeremylenz !

File: webpack/JobWizard/steps/Schedule/ScheduleReccuring.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    if (!validEnd || !repeatValid) {
+      setValid(false);
+    } else if (isFuture && startsAt.length) {
+      setValid(true);
+    } else if (!isFuture) {
+      setValid(true);
+    } else {
+      setValid(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (jeremylenz) [reply webpack/JobWizard/steps/Schedule/ScheduleReccuring.js]: @MariaAga This article is what I was trying to explain.. Dan explains it much better here

https://beta.reactjs.org/learn/you-might-not-need-an-effect

**Pull request merged**